src/components/EditSections/EditUserInfo/EditUserInfo.js

```diff
@@ -316,6 +340,20 @@ class EditUserInfo extends React.Component {
                 fullWidth
               />
             </Col>
+            <Col xs={12} md={3}>
+              <Field
+                label={<FormattedMessage id="ui-users.information.userType" />}
+                name="type"
+                id="userType"
+                component={Select}
+                fullWidth
+                defaultValue={initialValues.type}
```

Comment:
Probably it should be defined on `initialValues` level, depending on form status (create or edit) - https://final-form.org/docs/react-final-form/types/FieldProps#defaultvalue